src/main/java/io/camunda/zeebe/process/test/testengine/GrpcToLogStreamGateway.java

```diff
@@ -94,16 +94,20 @@ public GrpcToLogStreamGateway(
 
   private void writeCommandWithKey(
       final Long key, final RecordMetadata metadata, final BufferWriter bufferWriter) {
-    writer.reset();
+    synchronized (writer) {
```

Comment:
:x: This is not necessary. It is accessed by one single thread.

Reply:
Is it? Because after I added this I have not seen the issue anymore. So I was under the impression that this was the most important part of the fix.

Reply:
Maybe I oversee something but these are private methods which are always called in the executor, which is single threaded.

Reply:
You're right, I can't figure out why but I tested it again and this clearly fixes the issue. I'm gonna merge it and dig deeper into the why this would change anything later.

Reply:
@remcowesterhoud I think the problem is this method https://github.com/camunda-cloud/zeebe-process-test/blame/a10f59519ab4440f2f76d0e2156086c87fa9e7c3/src/main/java/io/camunda/zeebe/process/test/testengine/GrpcToLogStreamGateway.java#L284-L302